titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/env/TitusProvidedContainerEnvFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.master.mesos.kubeapiserver.direct.env;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.netflix.titus.api.jobmanager.model.job.BatchJobTask;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+
+public class TitusProvidedContainerEnvFactory implements ContainerEnvFactory {
+
+    private static final ContainerEnvFactory INSTANCE = new TitusProvidedContainerEnvFactory();
+
+    @Override
+    public Map<String, String> buildContainerEnv(Job<?> job, Task task) {
+        Map<String, String> env = new HashMap<>();
+        env.put("TITUS_JOB_ID", task.getJobId());
```

Comment:
are there constants for these somewhere?

Reply:
No, this is a block of code that I moved from another place. I can create constants for it.